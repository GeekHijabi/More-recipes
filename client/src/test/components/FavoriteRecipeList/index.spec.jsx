import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ConnectedFavoriteRecipes, { FavoriteRecipes }
  from '../../../js/components/FavoriteRecipeList';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


const props = {
  apiGetFavoriteRecipe: jest.fn(() => Promise.resolve({ ok: true })),
  favorites: []
};


describe('FavoriteRecipes snapshot', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<FavoriteRecipes {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Connected FavoriteList component', () => {
  it('component successfully rendered', () => {
    const store = mockStore({
      recipe: {
        favorites: []
      }
    });
    const wrapper = shallow(<ConnectedFavoriteRecipes store={store} />);
    expect(wrapper.length).toBe(1);
  });
});
