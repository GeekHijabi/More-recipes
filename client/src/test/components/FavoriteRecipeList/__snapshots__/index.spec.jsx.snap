// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteRecipes snapshot should render correctly 1`] = `
<div>
  <withRouter(Connect(RecipeHeader)) />
  <div
    style={
      Object {
        "margin": "0 auto",
        "minHeight": "760px",
        "width": "85%",
      }
    }
  >
    <div
      className="col-xs-12 col-sm-12 col-md-12 col-lg-12 heading"
    >
      <h4>
        <span>
          My Favorite Recipes
        </span>
      </h4>
    </div>
    <div
      className="row"
    >
      <span
        className="styleText"
      >
        You do not have any favorite recipe yet!
      </span>
    </div>
  </div>
  <Footer />
</div>
`;
